### DISCUSSION TITLE
Dynamically loading class libraries

### DISCUSSION THREAD

**Discussion by aniljbhatia** [AvaloniaUI/Avalonia#7434] [Q&A]
I have a class library with user controls and need to be able to load them dynamically.

It works fine when I add the namespace into the view but how can I add the namespace dynamically?

**Comment by hailstorm75** [comment]
If loading on application start-up is fine for you, you can check out my project for loading views, view models via dependency injection: https://github.com/hailstorm75/MarkDoc.Core

**Reply by hailstorm75** [reply]
The main application loads a list of plugins that have their specific views for configuring them. These views, and their respective view models, are not referenced by the main application. Plugins are discovered and loaded on application startup.
